compiler/rustc_passes/src/dead.rs

```diff
@@ -734,6 +740,15 @@ impl<'tcx> DeadVisitor<'tcx> {
             return;
         };
         let tcx = self.tcx;
+
+        let first_hir_id = tcx.hir().local_def_id_to_hir_id(first_id);
+        let first_lint_level = tcx.lint_level_at_node(lint::builtin::DEAD_CODE, first_hir_id).0;
+        assert!(dead_codes.iter().skip(1).all(|id| {
+            let hir_id = tcx.hir().local_def_id_to_hir_id(*id);
+            let level = tcx.lint_level_at_node(lint::builtin::DEAD_CODE, hir_id).0;
+            level == first_lint_level
+        }));
```

Comment:
This `warn_multiple_dead_codes` method seems like it would be a fairly cold code path (as in, you'd have to have a project with a lot of dead code to hit this method a lot), so it seemed fine to add another loop here.